docs/madr/decisions/056-meshservice-migration.md

```diff
@@ -0,0 +1,165 @@
+# MeshService migration
+
+* Status: accepted
+
+## Context and Problem Statement
+
+This MADR addresses the switch from not having MeshServices at all to MeshServices
+being automatically created, either from `Services` in Kubernetes or from
+`Dataplanes` in Universal.
+
+When upgrading from 2.8 to 2.9, nothing at the data plane level should change.
+But MeshService behaves differently in several different ways.
+
+In general the migration process for MeshService consists of no longer using
+`kuma.io/service` so:
+
+- targeting policies to the real `MeshService` resource
+- switching over to the `MeshService` i.e. `HostnameGenerator` hostnames, if not
+  using Kubernetes DNS
+
+### Policy
+
+If for example we have a policy targeted at:
+
+```
+spec:
+  targetRef:
+    kind: MeshService
+    name: demo-app_kuma-demo_svc_5000
+```
+
+when switching over to MeshService, this policy will no longer apply.
+
+### Traffic
```

Comment:
I think it can also affect `reachableServices`. If we have old `kuma.io/services` and `MeshService` outbounds, a user will now have outbounds included by reachable services annotation and MeshService one.

Reply:
Let's say that you have reachableServices defined, and you enable MeshServices. Your dp is going to observe all services instead of the one that you provided in rechableServices

Reply:
why? I think it will just stay as is. DPP will still be able to connect to the service using servicetag vip